### ISSUE TITLE
Bug in test_frcnn.py

### ISSUE THREAD

**Issue by Abhijit-2592** [yhenon/keras-frcnn#202]
In your test_frcnn.py in lines 
131 and 133 you have defined
```
model_classifier_only = Model([feature_map_input, roi_input], classifier)
model_classifier = Model([feature_map_input, roi_input], classifier)
```
Both are exactly the same models

But you are loading weights only for model_classifier but you are using model_classifier_only to predict the output in line 192. essentially you are predicting output with random weights.

The same bug is also present in measure_MAP.py

**Comment by aupilot** [comment]
The funny thinks that this classifier with random weights actually _works_  :)
Perhaps worse that a proper one though

**Comment by Abhijit-2592** [comment]
@aupilot  yeah you are right even I am surprised that this actually works. @yhenon  could you please clarify on this

**Comment by aupilot** [comment]
Someone (perhaps Le Cun?) have shown in an early paper that you can replace few layers with random weights and it will only slightly degrade performance. That's a common "problem" with NN - despite bugs they still works :)

**Comment by Abhijit-2592** [comment]
@aupilot is it? I am surprised! could you please give a link to that paper? It sounds interesting.

**Comment by oak-tree** [comment]
Hey all, 
I do not think it matter which of the `Model` you use. As those model share the same `layers` so they have the same `weights`

**Comment by Abhijit-2592** [comment]
@oak-tree I disagree with you, It is not about the base_nn (vgg or resnet) it is about the final_classifier thus there is no weight sharing there.

**Comment by giladdiv** [comment]
Its the not the same model but those are the same layers and when you load the weights in one model the other one change as well
You can see it by debugging and view model.get_weights() before and after

**Comment by oak-tree** [comment]
@Abhijit-2592 

Just like @giladdiv wrote: 
https://github.com/yhenon/keras-frcnn/blob/237bcc1e17e298456565ae86bea0279650497689/test_frcnn.py#L131-L133

Those two model have been created from the same `layer's instances' therefor share the same weights.

**Comment by apacha** [comment]
So why exactly do we need `model_classifier_only` at all? I just replaced
`[P_cls, P_regr] = model_classifier_only.predict([F, ROIs])` with `[P_cls, P_regr] = model_classifier.predict([F, ROIs])` and removed the variable entirely. Everything still works fine...

**Comment by FabricioProjects** [comment]
Hey all! Are you training with VGG-16 or Resnet50? When I train with VGG after 100 epochs using 1000 images, i only get 'bg' as the most probably class... can you please share yours training setup?

**Comment by Abhijit-2592** [comment]
@FabricioProjects  try this [repo](https://github.com/Abhijit-2592/Keras_object_detection). Yes I trained it on VGG16 (custom dataset) and I got god results. probably something wrong with training

**Comment by apacha** [comment]
@FabricioProjects I've used it in [this repo](https://github.com/apacha/MusicObjectDetector), but we switched to [Tensorflow Object Detection API](https://github.com/tensorflow/models/tree/master/research/object_detection), because it works MUCH better and allows also to test other object detection implementation (R-FCN, SSD).

As stated by the authors, this repo is now deprecated.

**Comment by FabricioProjects** [comment]
@Abhijit-2592 and @apacha , tks for the response. Do you think that 100 epochs using 1000 images is enough to show some results (total loss is about 1.77)? and do you think that 'bg' being the most probably class on prediction is a normal behavior? The dataset dont even has an 'bg' class training example... 
this piece of code doesnt let any class prediction pass through because (P_cls.shape[2] - 1) is the "bg" class:
if np.max(P_cls[0, ii, :]) < bbox_threshold or np.argmax(P_cls[0, ii, :]) == (P_cls.shape[2] - 1):
       continue
